game/world/objects/npc.cpp

```diff
@@ -2357,7 +2357,11 @@ void Npc::nextAiAction(AiQueue& queue, uint64_t dt) {
       break;
       }
     case AI_Attack:
-      if(currentTarget!=nullptr && weaponState()!=WeaponState::NoWeapon){
+      if(weaponState()==WeaponState::NoWeapon) {
+        if(!drawWeaponMelee())
+          break;
```

Comment:
Not 100% about this `break`

Basically you say, if weapon can't be drawn (`npc` is in water for example), then command should be skipped.
Skipping command mean `npc` wont attack(fine), but also won't chaise and move resulting in soft-lock in water.

Maybe it's better to move `drawWeaponMelee` part into `implAttack` ?